app/gui2/src/components/GraphEditor/GraphNodeMessage.vue

```diff
@@ -1,35 +1,50 @@
 <script setup lang="ts">
 import SvgIcon from '@/components/SvgIcon.vue'
 import type { Icon } from '@/util/iconName'
-import { computed } from 'vue'
-
-/** The type of a message. */
-export type GraphNodeMessageType = 'error' | 'warning'
 
 const props = defineProps<{
   message: string
   type: GraphNodeMessageType
-  icon?: Icon
 }>()
-
-const icon = computed(() => iconForType[props.type])
 </script>
 
 <script lang="ts">
-const styleClassForType: Record<GraphNodeMessageType, string> = {
-  error: 'GraphNodeError',
-  warning: 'GraphNodeWarning',
-}
+type MessageType = 'error' | 'warning' | 'panic'
+/** The type of a message. */
+export class GraphNodeMessageType {
+  private constructor(private readonly type: MessageType) {}
+
+  static Error: GraphNodeMessageType = new GraphNodeMessageType('error')
+  static Warning: GraphNodeMessageType = new GraphNodeMessageType('warning')
+  static Panic: GraphNodeMessageType = new GraphNodeMessageType('panic')
+
+  get iconName() {
+    return iconForMessageType[this.type]
+  }
 
-const iconForType: Record<GraphNodeMessageType, Icon | undefined> = {
+  get cssColor() {
+    return colorForMessageType[this.type]
+  }
+}
```

Comment:
Do we actually need a separate _class_ for it? I would stay with just `MessageType` and exported constants below.